docs/api-operation.md

```diff
@@ -60,6 +45,34 @@ const resizeThenRotate = await sharp(input)
 ```
 
 
+## autoOrient
+> autoOrient() ⇒ <code>Sharp</code>
+
+Auto-orient based on the EXIF `Orientation` tag, then remove the tag.
```

Comment:
It occurs to me now that I am trying to test this new version of Sharp with `@sveltejs/enhanced-img` that the metadata is not adjusted in time for some useful operations to take place. For example:

```
sharp(img, {autoOrient: true]}).metadata()
```

The above will show the metadata before any auto-orientation takes place, which seems like a headache from the perspective of someone using the tool. If I am asking sharp to auto-orient the image, why would I want the old width and height?

I see several ways to work this issue:

1. No change. All metadata is from the input state of the image.
2. If `autoOrient` is true, drop "orientation" and swap width and height as necessary from metadata.
3. Add an option to `.metadata({ autoOrient: boolean })` that opts you in to view the adjusted metadata.
4. Export a function `autoOrientMetadata(metadata)` that devs can use as needed on metadata objects.

I lean toward option 3, with an eye toward changing the default behavior at the next major.

No matter which path you'd like to take, I think any transforms on the metadata ought to also include the original stuff as well. Maybe something like the following:

```diff
{
  format: 'jpeg',
- width: 4032,
+ width: 3024,
- height: 3024,
+ height: 4032
  space: 'srgb',
  channels: 3,
  depth: 'uchar',
  density: 72,
  chromaSubsampling: '4:2:0',
  isProgressive: false,
  resolutionUnit: 'inch',
  hasProfile: false,
  hasAlpha: false,
- orientation: 6,
+ preAutoOrientation: {
+   orientation: 6,
+   width: 4032,
+   height: 3024
}
```

Reply:
I like the idea of adding extra fields to the metadata that include the dimensions after applying orientation, which is closest to option 5 here. There have been a few people ask questions along the lines of "why are the metadata dimensions not what I expect?" and this would help answer it.